### ISSUE TITLE
Copy/Paste from Notepad++ to cmd window loses line endings

### ISSUE THREAD

**Issue by ronww** [notepad-plus-plus/notepad-plus-plus#14688]
![CopyPasteLosesLineEndings](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/62124/b1e2bc05-7f34-4169-b0c5-bf5fb1bf4589)
### Description of the Issue
When copying multiple lines from Notepad++ and pasting into a cmd window, line endings are lost.
This does not happen when copying from other applications.

### Steps to Reproduce the Issue
1. Copy multiple lines in Notepad++
2. Paste into a cmd window
3. Observe that the lines are pasted as a single line

### Expected Behavior
Lines pasted into a cmd window are separate lines

### Actual Behavior
lines are pasted as a single line

### Debug Information
Notepad++ v8.5.3   (64-bit)
Build time : May 15 2023 - 06:09:36
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : "C:\Work\ParamTest.bat"
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 10 Enterprise (64-bit)
OS Version : 22H2
OS Build : 19045.3930
Current ANSI codepage : 1252
Plugins :
    mimeTools (2.9)
    NppConverter (4.5)
    NppExport (0.4)

**Comment by alankilborn** [comment]
Can't replicate under 8.6.2.

Multiple lines copied from Notepad++ paste just fine into a cmd window:

![image](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/77065706/10e77c6e-815c-46c7-812f-b6fdcd9a612d)

leads to this in the cmd window after attempting to paste:

![image](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/77065706/460e8a34-a511-42c3-82ab-65865684a78a)

**Comment by AndiKiissel** [comment]
This happens if End-Of-Line is not Windows CRLF:

![image](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/107611105/b86d97ad-4217-4c6c-97a6-6e4af0e3c82d)

**Comment by alankilborn** [comment]
@AndiKiissel 

>This happens if End-Of-Line is not Windows CRLF

For me, I get the same Warning prompt as I got before, even with LF line endings, or CR line endings in my Notepad++ tab.

**Comment by rdipardo** [comment]
> For me, I get the same Warning prompt as I got before, even with LF line endings, or CR line endings in my Notepad++ tab.

Now try the legacy command prompt that @AndiKiissel [is using](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/14688#issuecomment-1932670384), not the Windows Terminal app

1. <kbd>&#8862;</kbd> (Windows key) + <kbd>R</kbd>
3. Run `conhost.exe`

**Comment by alankilborn** [comment]
> Now try the legacy command prompt

OK, but I was merely "playing along" anyway...

Do we really care that Notepad++ is copying its exact data to the clipboard?
Is there some sort of requirement that data copied to the clipboard is provided with Windows' CRLF line-endings?

**Comment by AndiKiissel** [comment]
I only showed how it can happen, I don't think it needs to be changed.

**Comment by ronww** [comment]
I ran a test script in a cmd window, trying it with different line endings. When I pasted from other applications, line endings were present. But pasting from Notepad++, there were none.

But, apparently is an issue with the older version of Notepad++ that my employer has deployed to our PCs.

**Comment by alankilborn** [comment]
It would still be a "problem" with any version of Notepad++.
But the question remains:  If it is a typical _Copy_ to clipboard, and not some sort of _Copy Special_, should LF-only (or CR-only) line endings be converted to CR+LF, at the time of the copy?

This is not a problem when pasting into Notepad++, because Scintilla paste functions automatically convert incoming line-endings to be appropriate for the current tab's line-ending format.

So the argument is:  Notepad++ is liberal in what it accepts (converting as necessary) but not so strict in what it sends -- should it be equally-friendly BOTH ways?  Maybe I'm stating it poorly, but see [Postel's Law](https://en.wikipedia.org/wiki/Robustness_principle).

**Comment by ronww** [comment]
After reading alankilborn's post, I did some more experiments on my personal PC (after installing Notepad++ 8.6.2).
Turns out the problem is that cmd is not just ignoring the non-CRLF line endings, but discarding them.
Eclipse, FireFox and LibreOffice accept pastes with LF line endings. I don't know about other applications.
So, other than possibly adding a "special copy" feature, this turns out to not be a Notepad++ issue.
